test/test_runtime.py

```diff
@@ -444,6 +447,70 @@ def test_require_collection(runtime_tmp: Runtime) -> None:
     runtime_tmp.require_collection("community.molecule", "0.1.0")
 
 
+@dataclass
+class ScanSysPath:
+    """Parameters for scan tests."""
+
+    isolated: bool
+    scan: bool
+    expected: bool
+
+    def __str__(self) -> str:
+        """Return a string representation of the object."""
+        parts = [
+            f"{field.name}{str(getattr(self, field.name))[0]}" for field in fields(self)
+        ]
+        return "-".join(parts)
+
+
+@pytest.mark.parametrize(
+    ("param"),
+    (
+        ScanSysPath(isolated=True, scan=True, expected=False),
+        ScanSysPath(isolated=True, scan=False, expected=False),
+        ScanSysPath(isolated=False, scan=True, expected=True),
+        ScanSysPath(isolated=False, scan=False, expected=False),
```

Comment:
Based on the result, if `scan` is False, no matter what value `isolated` is, the `expected` is always False.

If `scan` is `True`, I think the `expected` should be True. It's not related to isolated.

Reply:
```if not self.isolated and self.config.collections_scan_sys_path:```

this one confirms that if isolated and scan sys is true we still don't inject

```
        ScanSysPath(isolated=True, scan=False, expected=False),
```

Reply:
I need @ssbarnea's input here, since lint uses isolated, I think he will better know the impact...  Maybe a sperate PR to remove the isolated logic and see what he says?

I do not have a good understanding of what isolated is intended to accomplish :/